android/src/toga_android/window.py

```diff
@@ -137,8 +146,68 @@ def get_visible(self):
     # Window state
     ######################################################################
 
-    def set_full_screen(self, is_full_screen):
-        self.interface.factory.not_implemented("Window.set_full_screen()")
+    def get_window_state(self):
+        # `window.state` is called in `_close()`, which itself is
+        # sometimes called during certain stages when the app
+        # attribute may not exist. In such cases, return NORMAL.
+        if getattr(self, "app", None) is None:
+            return WindowState.NORMAL
+        decor_view = self.app.native.getWindow().getDecorView()
+        system_ui_flags = decor_view.getSystemUiVisibility()
+        if system_ui_flags & (
+            decor_view.SYSTEM_UI_FLAG_FULLSCREEN
+            | decor_view.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+            | decor_view.SYSTEM_UI_FLAG_IMMERSIVE
+        ):
+            if self._in_presentation_mode:
+                return WindowState.PRESENTATION
+            else:
+                return WindowState.FULLSCREEN
+        return WindowState.NORMAL
+
+    def set_window_state(self, state):
+        current_state = self.get_window_state()
+        decor_view = self.app.native.getWindow().getDecorView()
+
+        if current_state == state:
+            return
+
+        elif current_state != WindowState.NORMAL:
+            if current_state == WindowState.FULLSCREEN:
+                decor_view.setSystemUiVisibility(0)
+
+            else:  # current_state == WindowState.PRESENTATION:
+                decor_view.setSystemUiVisibility(0)
+                self.show_actionbar(True)
+                self._in_presentation_mode = False
+
+            self.set_window_state(state)
+
+        else:  # current_state == WindowState.NORMAL:
+            if state == WindowState.MAXIMIZED:
+                # no-op on Android.
+                pass
+
+            elif state == WindowState.MINIMIZED:
+                # no-op on Android.
+                pass
+
+            elif state == WindowState.FULLSCREEN:
+                decor_view.setSystemUiVisibility(
+                    # These constants are all marked as deprecated as of API 30.
+                    decor_view.SYSTEM_UI_FLAG_FULLSCREEN
+                    | decor_view.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+                    | decor_view.SYSTEM_UI_FLAG_IMMERSIVE
+                )
+
+            else:  # state == WindowState.PRESENTATION:
+                decor_view.setSystemUiVisibility(
+                    decor_view.SYSTEM_UI_FLAG_FULLSCREEN
+                    | decor_view.SYSTEM_UI_FLAG_HIDE_NAVIGATION
+                    | decor_view.SYSTEM_UI_FLAG_IMMERSIVE
+                )
+                self.show_actionbar(False)
+                self._in_presentation_mode = True
```

Comment:
It can be set by `show_actionbar()`, but it would become confusing, since a state related attribute will be set in a relatively state-unrelated method. Setting the attribute in the state method itself, seems a bit clear as this pattern is also followed in other backends also.